parcels/include/parcels.h

```diff
@@ -94,6 +146,194 @@ static inline ErrorCode spatial_interpolation_tracer_c_grid_3D(int xi, int yi, i
   *value = data[zi][yi+1][xi+1];
   return SUCCESS;
 }
+static inline ErrorCode spatial_interpolation_tracer_c_grid_3D2(float data[2][2][2], float *value)
+{
+  *value = data[0][1][1];
+  return SUCCESS;
+}
+
+static inline int getBlock3D(int *chunk_info, int ti, int yi, int xi, int *block, int *index_local)
+{
+  int ndim = chunk_info[0];
+  if (ndim != 3)
+    exit(-1);
+  int chunksize, i;
+  int shape[ndim];
+  int index[3] = {ti, yi, xi};
+  for(i=0; i<ndim; ++i){
+    shape[i] = chunk_info[ndim+1+i];
+    chunksize = chunk_info[1+i];
+    block[i] = (int) index[i] / chunksize;
```

Comment:
If you apply a periodic halo to the fieldset, you will probably get uneven chunking along a dimension. For example, my `lon` dimension had block sizes of `5, 260, ..., 260, 5`. Calculating the block using the nominal "chunksize" doesn't work in this case. Instead, I put in a loop over the block sizes:
```suggestion
    int chunk_sum = 0, shift = 0;
    // shift index into blocksizes
    for (j = 0; j < i; j++) shift += chunk_info[1+ndim+j];

    for (j = 0; j < shape[i]; j++) {
      chunk_sum += chunk_info[1+2*ndim+shift+j];
      if (index[i] < chunk_sum) {
        chunk_sum -= chunk_info[1+2*ndim+shift+j];
        break;
      }
    }
    block[i] = j;
    index_local[i] = index[i] - chunk_sum;
```

A small loop over blocks is probably not a huge loss of efficiency. Something is still blowing memory usage up significantly, so I can't test further after applying this change...

Reply:
You are right. I'll clean that such that chunk indexing to take into account halos in particular and non-constant chunksizes in general.

Reply:
Thanks @angus-g, I'm doing those changes
+ `tdim` is not in the chunks anymore, so we don't look at it
+ `self.data.chunksize` is not used anymore, so it's removed from the `chunk_info` list